src/main/java/shortestpath/pathfinder/VisitedTiles.java

```diff
@@ -0,0 +1,61 @@
+package shortestpath.pathfinder;
+
+import net.runelite.api.coords.WorldPoint;
+
+public class VisitedTiles {
+    private static final int TOTAL_PLANES = 4;
+    private static final int REGION_SIZE = 64;
+
+    // TODO: what is the max number of regions?
+    private static final int REGION_EXTENT_X = 192;
+    private static final int REGION_EXTENT_Y = 256;
+    private final VisitedRegion[] visitedRegions = new VisitedRegion[REGION_EXTENT_X * REGION_EXTENT_Y];
```

Comment:
The region extent X and Y can be calculated inside `CollisionMap.fromResources()`, stored in static variables in `CollisionMap.java`, and referenced here as `CollisionMap.REGION_EXTENT_X` and `CollisionMap.REGION_EXTENT_Y`.
As for the theoretical maximum number of regions it is `6400 >> 6 = 100` for X (anything >= 6400 x are instanced areas which copy its collision map data from templates in < 6400 areas) and `2^14 >> 6 = 256` for Y. In the current collision map the calculated region extent X is 66 and Y is 196.

Reply:
I've opted to calculate the extents while loading the collision map and size `visitedTiles` to the smallest 2D array that stores all collision map data. It's now setup so if collision map data is ever downloaded/reloaded automatically at runtime, it should still be correct without having to adjust constants.